controllers/util/util.go

```diff
@@ -122,3 +127,74 @@ func GetSystemCerts() ([]byte, error) {
 	}
 	return data, err
 }
+
+func GetServiceHostname(ctx context.Context, svcName, ns string, client client.Client) (string, error) {
+	serviceHostname := ""
+	isAvailable, service, err := GetServiceIfAvailable(ctx, svcName, ns, client)
+	if err != nil {
+		return "", err
+	}
+
+	if isAvailable {
+		// Loop over all Service ports, if a secured port is found
+		// set port and scheme to its secured ones and exit the loop
+		servicePort := ""
+		scheme := "http"
+		for _, port := range service.Spec.Ports {
+			servicePort = fmt.Sprintf("%d", port.Port)
+			if servicePort == "8443" || servicePort == "443" {
+				// If a secured port is found, just set scheme to 'https://' and exit the loop
+				scheme = "https"
+				break
+			}
+		}
+		serviceHostname = scheme + "://" + service.Name + "." + service.Namespace + ".svc.cluster.local:" + servicePort
+	}
+	return serviceHostname, nil
+}
+
+func GetRouteHostname(ctx context.Context, routeName, ns string, client client.Client) (string, error) {
```

Comment:
In `dspipeline_params.go` we have
```
	if p.ExternalRouteEnabled(dsp) {
		route, err := p.RetrieveAndSetExternalRoute(ctx, client, log)
		if err != nil {
			log.Info("Unable to retrieve route", "error", err)
		}
		p.ObjectStorageConnection.ExternalRouteURL = route.Spec.Host
		p.ObjectStorageConnection.Endpoint = route.Spec.Host
		p.ObjectStorageConnection.Secure = util.BoolPointer(true)
		p.ObjectStorageConnection.Host = route.Spec.Host
		p.ObjectStorageConnection.Scheme = "https"
		//port should be empty when external route is specified
		p.ObjectStorageConnection.Port = ""
	}
```

Can we have a single method for that?

Reply:
I think these are different use cases. This function will update DSPA to get the route endpoint if, and only if, the endpoint is ready to serve requests. `dspipeline_params.go` serve to a different purpose.